syncer/syncer.go

```diff
@@ -1800,6 +1801,13 @@ func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext) e
 				}
 			}
 		}
+		if s.cfg.ShardMode == config.ShardOptimistic {
```

Comment:
How about changing `if` to `else if`? 😂

Reply:
doen in https://github.com/pingcap/dm/pull/905/commits/df32d37e8f3e8f0ca08f4fbdc7e62c39a4fc6c53